doc: explain using `lib.stylix.colors` (#328)
danth/stylix: docs/src/configuration.md

`stylix.base16Scheme` is not changed in the user config. If that is the case,
only the user override is used.

> ### Accessing

Reviewer: "Extending" may be more appropriate here.

```suggestion
### Extending
```

> When passing colors to unsupported targets, it is possible to access values from the configured

Reviewer: ```suggestion
If you are creating custom modules, it is possible to access values from the configured
```

Author: Some people prefer creating exports local to their config. I feel like limiting this suggestion to stylix modules might be deterring for them, which is why I phrased it like that.

Reviewer: That would be fine too.

> When passing colors to unsupported targets, it is possible to access generated
> color scheme through `lib.stylix.colors`. It contains an attrset, populated by

Reviewer: ```suggestion
When passing colors to unsupported targets, it is possible to access values from the configured
color scheme through `lib.stylix.colors`. It contains an attrset, populated by
```

The color scheme may have been be configured manually, rather than generated.

> color scheme through `lib.stylix.colors`. It contains an attrset, populated by
> [base16.nix](https://github.com/SenchoPens/base16.nix)'s `mkSchemeAttrs`.

Reviewer: ```suggestion
color scheme through `lib.stylix.colors`.
```

This sentence seems unnecessary since we link to `mkSchemeAttrs` below.

Author: I think it's better to state the origin of the value explicitly, as in "generated by", after the code example

> [base16.nix](https://github.com/SenchoPens/base16.nix)'s `mkSchemeAttrs`. TLDR
> on its contents is

Reviewer: How about this?

```suggestion
[base16.nix](https://github.com/SenchoPens/base16.nix)'s `mkSchemeAttrs`.
An overview of the available values is shown below.
```

```nix
lib.stylix.colors = {
  base08 = "ff0000";
  base08-hex-r = "ff";
  base08-dec-r = "0.996094";
  # ...
  red = "ff0000";
  # ...
  withHashtag = {
    base08 = "#ff0000";
    # ...
  };
};
```

> See `colors` function of `base16.nix` for more info.

Reviewer: It may also be useful to link to the function or at least the `base16.nix` repository.

When using [`stylix.override`](#overriding), unmodified color scheme can be
accessed from `lib.stylix.scheme` - which is otherwise identical to

> `lib.stylix.colors`.

Reviewer: `config.lib.stylix.scheme` was added as a workaround to try and retain the generated palette file after running `nix-collect-garbage`, and AFAIK was never meant to be public.

Author: I see. This probably deserves a comment in the source code, since by itself the purpose of the attribute is confusing

Reviewer: I believe that is what this comment a few lines below is referring to:

> generated.json is necessary in the presence of overrides.

Although it's not clear why `lib.stylix.scheme` is exported, when it could be defined as a local variable and only used to generate that file.

Author: Either way, this should be figured out separately from this PR
